spec.emu

```diff
@@ -0,0 +1,108 @@
+  <h2>Syntax</h2>
+  <emu-grammar type="definition">
+    ModuleBlockExpression :
+      `module` [no LineTerminator here] `{` ModuleBody? `}`
+  </emu-grammar>
+
+  <h2>Runtime Semantics: Evaluation</h2>
+  <emu-alg>
+    1. Let _moduleBlock_ be OrdinaryObjectCreate(%ModuleBlock.prototype%,  &laquo; [[HostDefinedSlot]], [[ModuleBlockBody]], [[SourceText]] &raquo;).
+    1. Set _moduleBlock_.[[SourceText]] to the source text matched by |ModuleBlockExpression|.
+    1. Perform ! HostInitializeModuleBlock(_moduleBlock_).
+    1. Return _moduleBlock_.
+  </emu-alg>
+</emu-clause>
+  
+<emu-clause id="sec-moduleblock">
+  <h1>ModuleBlock ( _body_ )</h1>
+  <p>The _body_ specifies the body (executable code) of a module block.</p>
+  <p>When the `ModuleBlock` function is called,the following steps are taken:</p>
+  <emu-alg>
+    1. If NewTarget is *undefined*
+      1. Throw a *TypeError* exception.
+    1. Let _parsedBody_ be ? ParseText(_body_, Module).
+    1. Let _moduleBlock_ be OrdinaryObjectCreateFromConstructor(%ModuleBlock.prototype%,  &laquo; [[HostDefinedSlot]], [[ModuleBlockBody]], [[SourceText]] &raquo;).
+    1. Let _sourceString_ be the string-concatenation of *"module {"*, _body_ and *"}"*.
+    1. Set _moduleBlock_.[[SourceText]] to _sourceString_.
+    1. Set _moduleBlock_.[[ModuleBlockBody]] to _parsedBody_.
+    1. Return _moduleBlock_.
+  </emu-alg>
+</emu-clause>
+
+<emu-clause id="sec-moduleblock.prototype">
+  <h1>ModuleBlock.prototype</h1>
+  <p>The initial value of `ModuleBlock.prototype` is the ModuleBlock prototype object.</p>
+  <p>This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *false* }.</p>
+</emu-clause>
+
+<emu-clause id="sec-properties-of-the-moduleblock-prototype-object">
+  <h1>Properties of the ModuleBlock Prototype Object</h1>
+  <p>The <dfn>ModuleBlock prototype object</dfn>:</p>
+  <ul>
+    <li>is <dfn>%ModuleBlock.prototype%</dfn>.</li>
+    <li>has a [[Prototype]] internal slot whose value is %Object.prototype%.</li>
+    <li>is an ordinary object.</li>
+  </ul>
+
+  <emu-clause id="sec-moduleblock.prototype.tostring">
+    <h1>ModuleBlock.prototype.toString ( )</h1>
+    <p>The following steps are taken:</p>
+    <emu-alg>
+      1. Let _module_ be the *this* value.
+      1. If Type(_module_) is Object and has the internal slot [[ModuleBlockBody]]
+        1. Return _module_.[[SourceText]].
+      1. Throw a *TypeError* exception.
+    </emu-alg>
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-import-calls">
+  <h1>Import Calls</h1>
+
+  <emu-clause id="sec-import-call-runtime-semantics-evaluation">
+    <h1>Runtime Semantics: Evaluation</h1>
+
+    <emu-grammar>ImportCall : `import` `(` AssignmentExpression `)`</emu-grammar>
+    <emu-alg>
+      1. Let _referencingScriptOrModule_ be ! GetActiveScriptOrModule().
+      1. Let _argRef_ be the result of evaluating |AssignmentExpression|.
+      1. Let _specifier_ be ? GetValue(_argRef_).
```

Comment:
Maybe `_specifierOrModule_`

Reply:
I like to think of module blocks as a kind of module specifier.